src/Console/CommandRunner.php

```diff
@@ -182,4 +182,16 @@ protected function getShell(ConsoleIo $io, CommandCollection $commands, $name)
 
         return $instance;
     }
+
+    /**
+     * The wrapper for creating shell instances.
+     *
+     * @param string $className Shell class name.
+     * @param \Cake\Console\ConsoleIo $io The io wrapper for the created shell class.
+     * @return \Cake\Console\Shell
```

Comment:
Not sure if I like that, though. Inheritance of some base logic can be useful. But out of scope of this discussion here now :)